cmd/juju/charmhub/convert.go

```diff
@@ -55,6 +55,15 @@ func convertInfoResponse(info transport.InfoResponse, arch string, risk charm.Ri
 	if err != nil {
 		return ir, errors.Trace(err)
 	}
+
+	if len(ir.Tracks) == 0 && len(ir.Channels) == 0 {
+		if ir.Charm != nil {
+			ir.Charm.Relations = nil
+		}
+
+		ir.NotFoundMessage = fmt.Sprintf("No charms have been published%s", buildFilterMessage(arch, risk, revision, track, base))
+	}
+
```

Comment:
This is presentation logic.
It should be located in the Print() method on the info writer.
All the info is there to be able to determine what message (if any) to print.